src/amm/erc_4626/mod.rs

```diff
@@ -17,61 +20,69 @@ use crate::{
     errors::{AMMError, ArithmeticError, EventLogError, SwapSimulationError},
 };
 
-use ethers::prelude::abigen;
-
 use super::uniswap_v2::{div_uu, q64_to_f64, U128_0X10000000000000000};
 
-abigen!(
-    IERC4626Vault,
-    r#"[
-        function totalAssets() external view returns (uint256)
-        function totalSupply() external view returns (uint256)
-        function decimals() external view returns (uint8)
-        event Withdraw(address indexed sender, address indexed receiver, address indexed owner, uint256 assets, uint256 shares)
-        event Deposit(address indexed sender,address indexed owner, uint256 assets, uint256 shares)
-
-    ]"#;
-);
+sol! {
+    /// Interface of the IERC4626Valut contract
+    #[derive(Debug, PartialEq, Eq)]
+    #[sol(rpc)]
+    contract IERC4626Vault {
+        event Withdraw(address indexed sender, address indexed receiver, address indexed owner, uint256 assets, uint256 shares);
+        event Deposit(address indexed sender,address indexed owner, uint256 assets, uint256 shares);
+        function totalAssets() external view returns (uint256);
+        function totalSupply() external view returns (uint256);
+        function decimals() external view returns (uint8);
+    }
+}
 
-pub const DEPOSIT_EVENT_SIGNATURE: H256 = H256([
+pub const DEPOSIT_EVENT_SIGNATURE: B256 = FixedBytes([
     220, 188, 28, 5, 36, 15, 49, 255, 58, 208, 103, 239, 30, 227, 92, 228, 153, 119, 98, 117, 46,
     58, 9, 82, 132, 117, 69, 68, 244, 199, 9, 215,
 ]);
 
-pub const WITHDRAW_EVENT_SIGNATURE: H256 = H256([
+pub const WITHDRAW_EVENT_SIGNATURE: B256 = FixedBytes([
     251, 222, 121, 125, 32, 28, 104, 27, 145, 5, 101, 41, 17, 158, 11, 2, 64, 124, 123, 185, 106,
     74, 44, 117, 192, 31, 201, 102, 114, 50, 200, 219,
 ]);
```

Comment:
Lets get rid of these and use `IERC4626Vault::Deposit::SIGNATURE` and `IERC4626Vault::Withdraw::SIGNATURE` instead.